core/src/test/java/org/elasticsearch/cluster/routing/allocation/ShardStateIT.java

```diff
@@ -48,8 +48,10 @@ public void testPrimaryFailureIncreasesTerm() throws Exception {
         indicesService.indexService(resolveIndex("test")).getShard(shard).failShard("simulated test failure", null);
 
         logger.info("--> waiting for a yellow index");
-        assertBusy(() -> assertThat(client().admin().cluster().prepareHealth().get().getStatus(),
-            equalTo(ClusterHealthStatus.YELLOW)));
+        // JDK 9 type inference gets confused, so we have to help the
+        // type inference
+        assertBusy((Runnable)(() -> assertThat(client().admin().cluster().prepareHealth().get().getStatus(),
```

Comment:
I usually find it cleaner to make a variable of type Runnable in cases like this. Doesn't matter though.